modules/library/render/src/test/java/org/geotools/renderer/lite/GridCoverageRendererTest.java

```diff
@@ -2400,6 +2408,78 @@ public void testContrastEnhancementWithNodataDataset() throws Exception {
         coverage.dispose(true);
     }
 
+    @Test
+    public void testSameCrs() throws Exception {
+        /**
+         * The test is checking that the rendering machinery is not applying the reprojection's
+         * extra steps when the requested map's CRS and the reader CRS are the same. There might be
+         * the case of a ProjectedCRS which has the baseCRS as lon/lat axis whilst the requested CRS
+         * is the same ProjectedCRS with same projection axis but with baseCRS having lat/lon axis.
+         */
+        CoordinateReferenceSystem utmProjection = CRS.decode("urn:ogc:def:crs:EPSG::25832", false);
+        String testLocation = "utm";
+        URL storeUrl = TestData.url(this.getClass(), testLocation);
+
+        File testDataFolder = new File(storeUrl.toURI());
+        File testDirectory = temporaryFolder.newFolder(testLocation);
+
+        File utmFile = new File(testDataFolder, "data.tif");
+        File destUtmFile = new File(testDirectory, "data.tif");
+        FileUtils.copyFile(utmFile, destUtmFile);
+
+        Hints hints = new Hints(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER, true);
+        ImageMosaicReader imReader = new ImageMosaicReader(testDirectory, hints);
+        ReferencedEnvelope mapExtent =
+                new ReferencedEnvelope(
+                        585173.1053764131,
+                        585989.3229074453,
+                        5670037.032323576,
+                        5670853.249854608,
+                        utmProjection);
+        Rectangle mapRasterArea = new Rectangle(768, 768);
+        AffineTransform worldToScreen =
+                RendererUtilities.worldToScreenTransform(mapExtent, mapRasterArea);
+        Interpolation interpolation =
+                InterpolationNearest.getInstance(Interpolation.INTERP_NEAREST);
+        RenderingHints interpolationHints =
+                new RenderingHints(JAI.KEY_INTERPOLATION, interpolation);
+        GridCoverageRenderer gcr =
+                new GridCoverageRenderer(
+                        mapExtent.getCoordinateReferenceSystem(),
+                        mapExtent,
+                        mapRasterArea,
+                        worldToScreen,
+                        interpolationHints);
+        gcr.setAdvancedProjectionHandlingEnabled(true);
+        gcr.setWrapEnabled(true);
+        RenderedImage ri = gcr.renderImage(imReader, null, null, interpolation, null, 256, 256);
+        RenderedOp ro = ((RenderedOp) ri);
+        Object rendering = ro.getRendering();
+
+        // Without the sameCrs fix, the involved processing chain would have been:
+        // ImageRead -> Scale -> Mosaic -> Scale -> Crop
+        // With the sameCrs fix, the involved processing chain is simply:
+        // ImageRead -> Scale -> Crop so that the final image is a MosaicOp
```

Comment:
The consequence seems less than clear to me. Is it because the Crop has been implemented using Mosaic? (I remember we sometimes do that, but cannot recall under which condition)

Reply:
Sorry. there was a copy-and-paste typo there.
I have also expanded a bit the comment.